_posts/2018-01-02-getting-up-and-running-with-react-native.md

```diff
@@ -0,0 +1,68 @@
+<br/>
+I planned to go through this [tutorial](https://facebook.github.io/react-native/docs/getting-started.html) on React Native website. In this post I am sharing my experience with the Getting Started section (yep, my post is actually longer than the Getting Started page haha).
+
+#### Step 1: install the tool create-react-native-app
+```
+npm install -g create-react-native-app
+```
+
+=> ✔️ Successful!
+
+#### Step 2: create a React Native app project
+```
+create-react-native-app HelloWorld
+```
+
+=> ❌ Unfortunately, failed! It didn't do much, just created a package.json file and then failed. The good news is that it failed with a very useful error message.
+
+![create-react-native-app error]({{ "/assets/blog/create-react-native-app-error.png" | absolute_url }}){:height="90%" width="90%"}
+
+I googled a bit trying to understand what was going on and why npm 5 wouldn't be supported by [create-react-native-app](https://github.com/react-community/create-react-native-app#installation). I haven't found the exact answer yet apart from `npm 5 is buggy`. The main suggestion to fix this problem is to downgrade npm to 4.x.x, but I don't really like the idea of downgrading, I think we should move forward, not backwards. So I searched a bit more and luckily came across a very useful comment on this [issue](https://github.com/react-community/create-react-native-app/issues/424):
+
+> when yarn is installed, CRNA already uses it by default to install packages.
+
+So I installed [yarn](https://yarnpkg.com/en/) (a package manager similar to npm) and it created my HelloWorld project! (Well it failed at the next step but that's okay, I feel like I haven't completely lost the battle...)
+
+
+#### Step 3: start the server
+
+```
+yarn start
+```
+
+=> ❌ Unfortunately, it failed too. But, again, with a very useful message.
+
+![yarn start error]({{ "/assets/blog/react-native-app-yarn-start-error.png" | absolute_url }}){:height="80%" width="80%"}
+
+I chose to install watchman (arbitrarily, I don't know what solution is best). However, when installing it, it errored:
+
+![brew install watchman error]({{ "/assets/blog/react-native-brew-watchman.png" | absolute_url }}){:height="55%" width="55%"}
+
+According to this [thread](https://stackoverflow.com/questions/29319378/cant-link-pcre-thru-brew-in-max-os-yosemite) on Stack Overflow, brew requires the content of the folder /user/local to be owned by you.
+```
+sudo chown -R $(whoiam) /usr/local/*
+brew link pcre
+```
+And that solved the problem, I was able to start the server, yay!
+
+Note that if you try to run `sudo chown -R $(whoiam) /usr/local` as suggested in the Stack Overflow thread, it fails on Mac High Sierra with `operation not permitted` (see [here](https://github.com/Homebrew/brew/issues/3228)). It looks like just the content, not the directory itself, must be owned by you (and can be own by you anyway).
```

Comment:
and can be own _ed_ by you anyway  ? - I'm not sure I understand what you mean here

Reply:
~content~ _contents_

Reply:
oops
👍

Reply:
Ah ok https://dictionary.cambridge.org/grammar/british-grammar/content-or-contents

Reply:
Assuming there's more than one thing in the directory, i.e files and folders then I would use the plural